### PR TITLE
Add gitignore

### PR DESCRIPTION
**Pull request by matthewfeickert**
The Jupyter gitignore
c.f. https://github.com/jupyter/notebook/blob/master/.gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,40 @@
+MANIFEST
+build
+dist
+_build
+docs/man/*.gz
+docs/source/api/generated
+docs/source/config.rst
+docs/gh-pages
+notebook/i18n/*/LC_MESSAGES/*.mo
```
Comment (lukasheinrich) [inline .gitignore]: the notebook/* lines are specific to the jupyter/notebook repo and probably should be removed

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+notebook/i18n/*/LC_MESSAGES/nbjs.json
+notebook/static/components
+notebook/static/style/*.min.css*
+notebook/static/*/js/built/
+notebook/static/*/built/
+notebook/static/built/
+notebook/static/*/js/main.min.js*
+notebook/static/lab/*bundle.js
+node_modules
```
Comment (lukasheinrich) [inline .gitignore]: since we don't have any javascript modules (yet), this can also be removed

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+*.egg-info
+*~
+*.bak
+.ipynb_checkpoints
+.tox
+.DS_Store
+\#*#
+.#*
+.coverage
+src
```
Comment (lukasheinrich) [inline .gitignore]: same as above

**Comment by lukasheinrich** [comment]
also can you rebase to the current master? it seems your branch still hast the old `.travis.yml`

**Comment by matthewfeickert** [comment]
@lukasheinrich unless you think we should also take out the `docs/*` lines then I think this is okay now.

**Comment by lukasheinrich** [comment]
I think the docs lines should be fine, since hopefully we'll have some actual docs as well :)

LGTM

**Pull request merged**